Use `file` to determine if the file is binary

rkiyanchuk opened the pull request:
`file` utility uses magic numbers to better determine if the file is binary, even if it doesn't have any extension.

The `file` utility is present in most distros by default, but if it isn't, the plugin falls back into old behavior (using file extension).

Also mixed tabs and spaces replaced with spaces only.

Commit:
Use `file` util to determine if file is binary

`file` determines file types based on magic numbers so it can successfully
tell if the file is binary even when it doesn't have any extension.

Commit:
Fixed mixed spaces and tabs

Replaced all tabs with spaces.

Commit:
Fix symlinks recognition

fidian commented:
I am a little concerned about this patch.  I would want the check for isBinary to be called only once when opening the file, not called again later.  Let's pretend that I open a binary file and change that file so it is text on disk without replacing the buffer.  How do these updates handle tricky situations like that?

I am not at a computer for a couple days otherwise I would try this out myself.

Commit:
Use IsBinary only on file opening

IsBinary is called and used to set `&binary` variable, later only
`&binary` is used.

rkiyanchuk commented:
@fidian You are right, I didn't pay much attention to how `&binary` was used in the original code.

How about the updated version? `IsBinary` is called at the beginning and the resulting value is assigned to `&binary`. If `file` executable is not present it just preserves `&binary` value that was already set. Later only `&binary` variable is used.

fidian commented:
Looks ideal to me.  I appreciate the contribution!
